GTFS_converter.py

```diff
@@ -0,0 +1,241 @@
+edges_clean = {f'Edge{i}': info for i, (_, info) in enumerate(unique_combinations.items(), start=1)}
+
+
+for i, route in enumerate(compiled_list_of_routes, start=0): #since routes was stored as a list, it's converted to dict here
+    route_key=ordered_list_of_trips[i]
+    routes[route_key]={}
+    for j, component in enumerate(route, start=1):
+        comp_key = f'component{j}'
+        u, v, key =component
+        routes[route_key][comp_key]={'u': u.split(':')[1], 'v': v.split(':')[1], 'key': int(key.split(':')[1])}
+
+
+#merge trips and stop_times to tie trips and blocks to departure times
+df_stoptimes = df_stoptimes.merge(df_trips[['trip_id', 'block_id']], on='trip_id', how='left')
+df_dict = df_stoptimes.to_dict('records')
+
+for record in df_dict: #loop to create a dict called tours which contains the trips and their departure times 
+    block_id = 'block' + str(record['block_id'])
+    trip_id = record['trip_id']
+    departure_time = {'departure': record['departure_time']}
+    if block_id not in tours:
+        tours[block_id] = {}
+    if trip_id not in tours[block_id]:
+        tours[block_id][trip_id] = []
+    tours[block_id][trip_id].append(departure_time)
+
+# This function converts time to seconds since midnight (useful in tours.json since departure times are seconds away from 0
+def time_to_seconds(t):
+    h, m, s = map(int, t.split(':'))
+    return h * 3600 + m * 60 + s
+
+formatted_tours = []
+
+for block, trips in tours.items():
+    block_id = block.replace('block', '')
+    block_dict = {
+        "name": block_id,
+        "creation_time": 0,
+        "deletion_time": 172800,
+        "routes": []
+    }
+    
+    for trip, times in trips.items():
+        args_list = []
+        for time in times:
+            args_list.append({"departure": time_to_seconds(time['departure'])})
+            args_list.append(None)  # Insert None between departures (since stations have departure times but tracks do not, and in GTFS, only stations are represented, so it's safe to add null between the departure times)
+        args_list.pop()  # Remove the last None to avoid having it after the last departure
+        for identical_trip_group in identical_trip_groups: #this condition checks if a trip is identical to another trip, then the first instance of the trip_id will be used for all the identical trips
+            if len(identical_trip_group)>1:
+                if trip in identical_trip_group[1:]:
+                    route_dict = {
+                        "name": identical_trip_group[0],
+                        "args": args_list
+                    }
+                    break
+                else:
+                    route_dict = {
+                        "name": trip,
+                        "args": args_list
+                    }
+        block_dict["routes"].append(route_dict)
+    
+    formatted_tours.append(block_dict)
+
+tours_json = json.dumps(formatted_tours, indent=2)
+with open('tours.json', 'w') as file:

```

Comment:
possible enhancement to allow user to specify a folder as I suspect one might run batches of this.